indico/modules/events/registration/schemas.py

```diff
@@ -126,7 +135,13 @@ def _get_registration_data(self, registration):
             # File field stores the uuid as data which is not helpful.
             # We want to show the filename instead.
             if field['id'] in filenames:
+                obj = registration.data_by_field[field['id']]
```

Comment:
`data_by_field` is a property that iterates over all the fields, I would call this outside the `for field in fields` loop to avoid `O(n²)` complexity

Reply:
Actually this is not needed at all, because the input type is already available via `field['inputType']`

Reply:
Actually, since this only runs for file fields and not for every single field, it's fine to keep it as-is.